File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.go
func evalRPN(tokens []string) int {
	nums := []int{}
	for i := range tokens {
		switch tokens[i] {
		case "+":
			x := nums[len(nums)-2] + nums[len(nums)-1]
			nums = nums[:len(nums)-2]
			nums = append(nums, x)
		case "-":
			x := nums[len(nums)-2] - nums[len(nums)-1]
			nums = nums[:len(nums)-2]
			nums = append(nums, x)
		case "*":
			x := nums[len(nums)-2] * nums[len(nums)-1]
			nums = nums[:len(nums)-2]
			nums = append(nums, x)
		case "/":
			x := nums[len(nums)-2] / nums[len(nums)-1]
			nums = nums[:len(nums)-2]
			nums = append(nums, x)
		default:
			n, _ := strconv.Atoi(tokens[i])
			nums = append(nums, n)
		}
	}
	return nums[0]
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack = Vec::<i32>::new();

        for t in tokens.iter() {
            match t.as_str() {
                "+" | "-" | "*" | "/" => {
                    let (b, a) = (stack.pop().unwrap(), stack.pop().unwrap());

                    match t.as_str() {
                        "+" => stack.push(a.saturating_add(b)),
                        "-" => stack.push(a.saturating_sub(b)),
                        "*" => stack.push(a.saturating_mul(b)),
                        "/" => stack.push(a.saturating_div(b)),
                        _ => (),
                    }
                }
                _ => stack.push(t.parse().unwrap()),
            }
        }

        stack.pop().unwrap()
    }
}
